File: app/src/main/java/com/medo/tweetspie/onboarding/OnboardingContract.java
package com.medo.tweetspie.onboarding;

import com.medo.tweetspie.base.BaseView;
import com.medo.tweetspie.base.BaseViewPresenter;
import com.twitter.sdk.android.core.Callback;
import com.twitter.sdk.android.core.TwitterSession;


public interface OnboardingContract {

  interface View extends BaseView {

    void setupTwitterButton(Callback<TwitterSession> callback);

    void exitWithError(String errorMessage);
  }


  interface Presenter extends BaseViewPresenter<View> {

    void onLoginSuccess(TwitterSession twitterSession);

    void onLoginFailure(Exception e);
  }
}

File: app/src/main/java/com/medo/tweetspie/onboarding/OnboardingContract.kt
package com.medo.tweetspie.onboarding

import com.medo.tweetspie.base.BasePresenter
import com.medo.tweetspie.base.BaseView
import com.twitter.sdk.android.core.Callback
import com.twitter.sdk.android.core.TwitterSession


interface OnboardingContract {

    interface View : BaseView {

        fun setupTwitterButton(callback: Callback<TwitterSession>)

        fun exitWithError(errorMessage: String)
    }


    interface Presenter : BasePresenter<View> {

        fun onLoginSuccess(twitterSession: TwitterSession)

        fun onLoginFailure(e: Exception)
    }
}